src/core/src/op/scatter_elements_update.cpp

```diff
@@ -78,256 +71,127 @@ shared_ptr<Node> op::v12::ScatterElementsUpdate::clone_with_new_inputs(const Out
-        TYPE_AXS_CASE(u32, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-        TYPE_AXS_CASE(u64, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-    default:
-        rc = false;
-        break;
-    }
-    return rc;
-}
-
-#define TYPE_IND_CASE(a, ...)                                      \
-    case element::Type_t::a: {                                     \
-        OV_OP_SCOPE(OV_PP_CAT3(scatter_element_update_ind, _, a)); \
-        rc = evaluate<DT, element::Type_t::a>(__VA_ARGS__);        \
-    } break;
-
-template <element::Type_t DT>
-bool evaluate(const ngraph::HostTensorPtr& arg0,
-              const ngraph::HostTensorPtr& arg1,
-              const ngraph::HostTensorPtr& arg2,
-              const ngraph::HostTensorPtr& arg3,
-              const ngraph::HostTensorPtr& out,
-              const int64_t normalized_axis,
-              const op::v12::ScatterElementsUpdate::Reduction reduction_type,
-              const bool use_init_value) {
-    auto indices_type = arg1->get_element_type();
-
-    // Dispatch specialization based on indicies data type.
-    bool rc = true;
-
-    switch (indices_type) {
-        TYPE_IND_CASE(i8, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-        TYPE_IND_CASE(i16, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-        TYPE_IND_CASE(i32, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-        TYPE_IND_CASE(i64, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-        TYPE_IND_CASE(u8, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-        TYPE_IND_CASE(u16, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-        TYPE_IND_CASE(u32, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-        TYPE_IND_CASE(u64, arg0, arg1, arg2, arg3, out, normalized_axis, reduction_type, use_init_value);
-    default:
-        rc = false;
-        break;
+struct Evaluate : public element::NoAction<bool> {
+    using element::NoAction<bool>::visit;
+
+    template <element::Type_t DATA_ET, class DT = fundamental_type_for<DATA_ET>>
+    static result_type visit(const Tensor& data,
+                             const Tensor& indices,
+                             const Tensor& updates,
+                             Tensor& output,
+                             const Shape& data_shape,
+                             const Shape& indices_shape,
+                             const int64_t normalized_axis,
+                             const op::v12::ScatterElementsUpdate::Reduction reduction_type,
+                             const bool use_init_value
+
+    ) {
+        using namespace ov::element;
+        return IfTypeOf<i8, i16, i32, i64, u8, u16, u32, u64>::apply<EvalByIdxType>(indices.get_element_type(),
```

Comment:
Should `EvalByIdxType` be renamed?

Reply:
`Eval` or `Evaluate`?
`ByIndex` or `ByIndices`?
My choice would be `EvaluateByIndicesType`. Note that this varies among other OPs, so let's decide offline common convention for it and apply it in separate PR. @praasz what's your opinion?

Reply:
@t-jankowski I saw you do such renaming in another PR, so it was kind reminder, not call to action 😄 
`EvaluateByIndicesType` sound nice to me

Reply:
In NonZero case second template bases on output's element type, so it had been mislead name so I changed it. I thought you regarded abbreviation, which should avoided in general.